### PR TITLE
fix barbarian spawning

### LINKED ISSUE #1688: Barbarians - Issues
minecolonies-universal-
minecraft 1.12.2-
mine colonies 0.8.4958

I'm experiencing several issues related to barbarians.

1.   Guard doesn't see raiders as a hostile mob.   I actual saw a guard walk right through barbarians go to their next waypoint on their patrol path.  Never reacted to the barbarians.   It appears their path is actually more important then a barbarians.  But  react as they should to a regular minecraft mob.

2.  Guards (my opinion) rush to protect citizens that are under attack, specially if they are not actively fighting anyone.

3.  Citizens reaction to raids are so wrong.   Had a raid and they didn't get they until morning while citizens were working.   It was a slaughter.  Have 12 citizens, when I got back into town to kill the barbarians.  There was 2 citizens left the guards.    The guards killed not one barbarians,  they where just patrolling like normal.   I saw 2 citizens as I was approaching that were not even running from the barbarians.   The barbarians where just standing in one place.   The 2 citizens appeared stuck in one tick to run away and would run away for a few block and then run back towards their working hut (like fish hut, farmer hut).  Then when they get a few blocks going the other way, turn back.   So they where just running back any forth.  Allowing barbarians to stand in one place hitting them as they run like a chicken with their heads cut off.
  Citizens should run back to their huts for safety or the town hall.

4. Guards didn't kill one barbarian,  I ended up killing them.

5.  You guys added a saying in the chat about who killing the guard and citizens.  That kill has issues.  when a guard killed the citizen, it just said the language text not the english text.   I don't remember it but it was the text for the guards name.  it was like  minecolonies.guard.name.   something like that appeared

### PR DESCRIPTION
**Pull request by Raycoms**
Closes #1688

# Changes proposed in this pull request:
- Fix barbarian spawning inside the colony.

Review please

### CODE REVIEW

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 47 issues
* ![MAJOR][MAJOR] 8 major
* ![MINOR][MINOR] 39 minor

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [Colony.java#](https://github.com/Minecolonies/minecolonies/blob/09aa54fcf0f0253d7f5bf65b05c5ad4d829f9eb4/src/main/java/com/minecolonies/coremod/colony/Colony.java#): This file has 1,387 lines, which is greater than 1,000 authorized. Split it into smaller files. [rule](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [Colony.java#L62](https://github.com/Minecolonies/minecolonies/blob/09aa54fcf0f0253d7f5bf65b05c5ad4d829f9eb4/src/main/java/com/minecolonies/coremod/colony/Colony.java#L62): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 51 to the maximum authorized 30 or less. [rule](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [Colony.java#L62](https://github.com/Minecolonies/minecolonies/blob/09aa54fcf0f0253d7f5bf65b05c5ad4d829f9eb4/src/main/java/com/minecolonies/coremod/colony/Colony.java#L62): class "Colony" has 100 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Colony.java#L640](https://github.com/Minecolonies/minecolonies/blob/09aa54fcf0f0253d7f5bf65b05c5ad4d829f9eb4/src/main/java/com/minecolonies/coremod/colony/Colony.java#L640): Reduce the number of returns of this method 11, down to the maximum allowed 6. [rule](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [Colony.java#L1091](https://github.com/Minecolonies/minecolonies/blob/09aa54fcf0f0253d7f5bf65b05c5ad4d829f9eb4/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1091): The Cyclomatic Complexity of this method "onWorldTick" is 20 which is greater than 15 authorized. [rule](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MAJOR][MAJOR] [Colony.java#L1534](https://github.com/Minecolonies/minecolonies/blob/09aa54fcf0f0253d7f5bf65b05c5ad4d829f9eb4/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1534): Assign this magic number 0.5D to a well-named constant, and use the constant instead. [rule](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1534](https://github.com/Minecolonies/minecolonies/blob/09aa54fcf0f0253d7f5bf65b05c5ad4d829f9eb4/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1534): Assign this magic number 0.1D to a well-named constant, and use the constant instead. [rule](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1534](https://github.com/Minecolonies/minecolonies/blob/09aa54fcf0f0253d7f5bf65b05c5ad4d829f9eb4/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1534): Assign this magic number 0.5D to a well-named constant, and use the constant instead. [rule](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS109)
1. ![MINOR][MINOR] [blocks#](https://github.com/Minecolonies/minecolonies/blob/09aa54fcf0f0253d7f5bf65b05c5ad4d829f9eb4/src/main/java/com/minecolonies/coremod/blocks#): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [gui#](https://github.com/Minecolonies/minecolonies/blob/09aa54fcf0f0253d7f5bf65b05c5ad4d829f9eb4/src/main/java/com/minecolonies/coremod/client/gui#): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1228)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Review by Asherslab — APPROVED** [review]

File: src/main/java/com/minecolonies/coremod/colony/Colony.java [hunk 1]
```diff
@@ -2124,8 +2124,8 @@ public void setCanBeAutoDeleted(final Boolean canBeDeleted)
 
     public BlockPos getRandomOutsiderInDirection(final EnumFacing directionX, final EnumFacing directionZ)
     {
-        final List<BlockPos> positions = wayPoints.keySet().stream().filter(pos -> isInDirection(directionX, directionZ, center.subtract(pos))).collect(Collectors.toList());
-        positions.addAll(buildings.keySet().stream().filter(pos -> isInDirection(directionX, directionZ, center.subtract(pos))).collect(Collectors.toList()));
+        final List<BlockPos> positions = wayPoints.keySet().stream().filter(pos -> isInDirection(directionX, directionZ, pos.subtract(center))).collect(Collectors.toList());
+        positions.addAll(buildings.keySet().stream().filter(pos -> isInDirection(directionX, directionZ, pos.subtract(center))).collect(Collectors.toList()));
```
Comment (marvin-bitterlich) [inline src/main/java/com/minecolonies/coremod/colony/Colony.java:2128]: if you flip it anyways, why not use a Math.abs on the result? That way you get the distance

Comment (Raycoms) [reply src/main/java/com/minecolonies/coremod/colony/Colony.java:2128]: because I want the vector in which direction it goes and not the distance =D

**Review by marchermans — APPROVED** [review]

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 47 issues
* ![MAJOR][MAJOR] 8 major
* ![MINOR][MINOR] 39 minor

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [Colony.java#](https://github.com/Minecolonies/minecolonies/blob/0409ea6780d43927880a7ce9ed5a21b3bf5d9fb2/src/main/java/com/minecolonies/coremod/colony/Colony.java#): This file has 1,387 lines, which is greater than 1,000 authorized. Split it into smaller files. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [Colony.java#L62](https://github.com/Minecolonies/minecolonies/blob/0409ea6780d43927880a7ce9ed5a21b3bf5d9fb2/src/main/java/com/minecolonies/coremod/colony/Colony.java#L62): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 51 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [Colony.java#L62](https://github.com/Minecolonies/minecolonies/blob/0409ea6780d43927880a7ce9ed5a21b3bf5d9fb2/src/main/java/com/minecolonies/coremod/colony/Colony.java#L62): class "Colony" has 100 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Colony.java#L640](https://github.com/Minecolonies/minecolonies/blob/0409ea6780d43927880a7ce9ed5a21b3bf5d9fb2/src/main/java/com/minecolonies/coremod/colony/Colony.java#L640): Reduce the number of returns of this method 11, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [Colony.java#L1091](https://github.com/Minecolonies/minecolonies/blob/0409ea6780d43927880a7ce9ed5a21b3bf5d9fb2/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1091): The Cyclomatic Complexity of this method "onWorldTick" is 20 which is greater than 15 authorized. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MAJOR][MAJOR] [Colony.java#L1534](https://github.com/Minecolonies/minecolonies/blob/0409ea6780d43927880a7ce9ed5a21b3bf5d9fb2/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1534): Assign this magic number 0.5D to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1534](https://github.com/Minecolonies/minecolonies/blob/0409ea6780d43927880a7ce9ed5a21b3bf5d9fb2/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1534): Assign this magic number 0.1D to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1534](https://github.com/Minecolonies/minecolonies/blob/0409ea6780d43927880a7ce9ed5a21b3bf5d9fb2/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1534): Assign this magic number 0.5D to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MINOR][MINOR] [blocks#](https://github.com/Minecolonies/minecolonies/blob/0409ea6780d43927880a7ce9ed5a21b3bf5d9fb2/src/main/java/com/minecolonies/coremod/blocks#): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [gui#](https://github.com/Minecolonies/minecolonies/blob/0409ea6780d43927880a7ce9ed5a21b3bf5d9fb2/src/main/java/com/minecolonies/coremod/client/gui#): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**